app/post.go

```diff
@@ -34,6 +34,25 @@ func CreatePostAsUser(post *model.Post) (*model.Post, *model.AppError) {
 		return nil, err
 	}
 
+	if utils.IsLicensed() && *utils.Cfg.TeamSettings.ExperimentalTownSquareIsReadOnly &&
+		channel.Name == model.DEFAULT_CHANNEL &&
+		!UserHasPermissionToExecute(post.UserId, model.PERMISSION_MANAGE_SYSTEM) {
+		err := model.NewLocAppError("createPost", "api.post.create_post.town_square_read_only", nil, "")
+		SendEphemeralPost(
+			channel.TeamId,
+			post.UserId,
+			&model.Post{
+				ChannelId: channel.Id,
+				ParentId:  post.ParentId,
+				RootId:    post.RootId,
+				UserId:    post.UserId,
+				Message:   utils.T("api.post.read_only_town_square"),
```

Comment:
This should use `utils.GetUserTranslations(userLocale)` to get the translation function for that user. Right now, it's translated to the system language, not the user one

Reply:
We would need to get the user to get the translation `T := utils.GetUserTranslations(user.Locale)`.
I propose we get the user here and pass it down to `CreatePost` and `handlePostEvents` as you comment here: https://github.com/mattermost/platform/pull/7140#discussion_r135810306

Reply:
@hmhealey We would need to get the user to get the translation T := utils.GetUserTranslations(user.Locale).
I propose we get the user here and pass it down to CreatePost and handlePostEvents as you comment here: #7140 (comment)